test/jdk/com/sun/net/httpserver/bugs/8300268/MaxIdleConnectionsTest.java

```diff
@@ -0,0 +1,141 @@
+        server.stop(0);
+    }
+
+    // Issue one too many requests and assert that the idle connection pool doesn't
+    // exceed maxIdleConnections
+    @Test
+    public void test() throws Exception {
+        final String host = server.getAddress().getHostString();
+        final int port = server.getAddress().getPort();
+
+        final List<Future<Void>> responses = new ArrayList<>();
+        try (final ExecutorService requestIssuer = Executors.newFixedThreadPool(totalConnections)) {
+            for (int i = 1; i <= totalConnections; i++) {
+                final URL requestURL = new URL("http://" + host + ":" + port + "/MaxIdleConnectionTest/" + i);
+                final Future<Void> result = requestIssuer.submit(() -> {
+                    System.out.println("Issuing request " + requestURL);
+                    final URLConnection conn = requestURL.openConnection();
+                    try (final InputStream is = conn.getInputStream()) {
+                        is.readAllBytes();
+                    }
+                    return null;
+                });
+                responses.add(result);
+            }
+            // wait for all the requests to reach each of the handlers
+            System.out.println("Waiting for all " + totalConnections + " requests to reach" +
+                    " the server side request handler");
+            reqFinishedProcessing.await();
+        }
+
+        // verify every request got served before checking idle count
+        for (int i = 0; i < totalConnections; i++) {
+            responses.get(i).get();
+            System.out.println("Received successful response for request " + i);
+        }
+
+        // assert that the limit set by maxIdleConnections was not exceeded
+        int idleConnectionCount = HttpServerAccess.getIdleConnectionCount(server);
+        System.out.println("count " + idleConnectionCount);
+        assertTrue(maxIdleConnections >= idleConnectionCount,
+                String.format("Too many idle connections: %d, limit: %d", idleConnectionCount, maxIdleConnections));
+    }
+
+    // Create HttpServer that will handle requests with multiple threads
+    private static HttpServer startServer(final CountDownLatch reqFinishedProcessing) throws IOException {
+        final var bindAddr = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
+        final HttpServer server = HttpServer.create(bindAddr, 0);
+
+        final AtomicInteger threadId = new AtomicInteger();
+        server.setExecutor(Executors.newCachedThreadPool(r -> {
+            final Thread t = new Thread(r);
+            t.setName("http-request-handler-" + threadId.incrementAndGet());
+            t.setDaemon(true);
+            return t;
+        }));
+
+        server.createContext("/MaxIdleConnectionTest/", (exchange) -> {
+            System.out.println("Request " + exchange.getRequestURI()
+                    + " received; handler will wait on latch");
```

Comment:
Hello Darragh, this looks like a typo here. This handler implementation doesn't wait on a latch (and instead counts it down). So I think removing the latter part of this log message will avoid any confusion.